File: backend/app/schemas/advisor.py
from typing import Optional, List
from pydantic import BaseModel

class NextBestActionResponse(BaseModel):
    lead_id: int
    lead_name: str
    current_status: str
    suggested_next_status: Optional[str] = None
    primary_action: str
    suggested_channel: str
    urgency: str
    talking_points: List[str]
    stage_progression_readiness: bool
    blockers: List[str]
    health_score: int
    health_category: str

File: frontend/src/types/advisor.ts
export interface NextBestAction {
  lead_id: number;
  lead_name: string;
  current_status: string;
  suggested_next_status?: string | null;
  primary_action: string;
  suggested_channel: string;
  urgency: 'Normal' | 'High' | 'Urgent';
  talking_points: string[];
  stage_progression_readiness: boolean;
  blockers: string[];
  health_score: number;
  health_category: string;
}
